src/Microsoft.AspNet.Mvc.TagHelpers/Internal/FileVersionProvider.cs

```diff
@@ -76,35 +77,39 @@ public string AddFileVersionToPath(string path)
                 return path;
             }
 
-            var fileInfo = _fileProvider.GetFileInfo(resolvedPath);
-            if (!fileInfo.Exists)
+            string value;
+            if (!_cache.TryGetValue(path, out value))
```

Comment:
This isn't new behavior, but this cache instance is shared with the user. We should use a better key than this to avoid collisions.